File: TP file d'entier/Code Java/src/File.java
import java.util.ArrayList;

public class File {
	private ArrayList<Integer> tabFile;
	private final static int MAX = 20;
	
	public File() {
		tabFile = new ArrayList<Integer>();
	}
	
	public synchronized String toString() {
		String chaineFile = "|File| = "+tabFile.size()+" :   <";
		for(int i = 0; i<tabFile.size(); i++) {
			if(i==tabFile.size()-1) { chaineFile+=Integer.toString(tabFile.get(i)); }
			else { chaineFile+=Integer.toString(tabFile.get(i))+", ";  }
		}
		chaineFile+="<";
		return chaineFile;
	}
	
	public synchronized String enfiler(int e) throws InterruptedException {
		while(tabFile.size()==MAX) wait();
		tabFile.add(e);
		notifyAll();
		return (" produit "+e+"\n");
	}
	
	public synchronized String defiler() throws InterruptedException {
		while(tabFile.size()==0) wait();
		int e = tabFile.get(0);
		tabFile.remove(0);
		notifyAll();
		return (" consomme "+e);
	}
	
	public synchronized void clear() throws InterruptedException {
		tabFile.clear();
	}

}

File: TP file d'entier/Code Python/File.py
# coding: utf-8

from threading import Condition

#________________Class File______________________________________________________________

class File :
    def __init__(self):
        self.tabFile=[]
        self.cond=Condition()

    
    #******_Methode_*****

    def toString(self):
        with self.cond:
            chaineFile = "|File| = " + str(len(self.tabFile))+" :   <"
            for i,e in enumerate(self.tabFile):
                if(i==len(self.tabFile)-1):
                    chaineFile+=str(self.tabFile[i])
                else:
                    chaineFile+=str(self.tabFile[i])+", "
            chaineFile+="<"
            return chaineFile

            

    def enfiler(self, e):
        with self.cond:
            while(len(self.tabFile)==20):
                self.cond.wait()
            self.tabFile.append(e)
            self.cond.notifyAll()
            return (" produit "+str(e)+"\n")

        

    def defiler(self):
        with self.cond:
            while(len(self.tabFile)==0):
                  self.cond.wait()
            e = self.tabFile[0]
            del self.tabFile[0]
            self.cond.notifyAll()
            return (" consomme "+str(e))

    def clear(self):
        with self.cond:
            self.tabFile.clear
